sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,279 @@
+__author__ = 'TDWilkinson'
+
+"""
+AIA response functions by integrating ChiantiPy
+
+    - analyze AIA instrument response that tells efficiency of channel using instrument properties
+    - then output a wavelength response
+    - use spectral model (synthetic spectra) - to get ion emissivities as a function of temp/density
+    - obtain temperature response based on chianti spectral contribution functions (emissivity)
+
+
+other important variables:
+effective area
+ion emissivity
+
+electron density
+temperature array
+
+# utilize sunpy/instr/tests/test_aia.py
+"""
+
+import numpy as np
+import pandas as pd
+from scipy.io import readsav
+
+import sunpy
+import sunpy.data.test as test
+
+
+def get_function(ion, emissivity, temperature, density, optional = None):
+    """
+    General format of functions where this is the statement of usefulness of function. If these were fits images, it would search the header for relevant information.
+
+    Parameters
+    ----------
+    variable1 : what module of import is used.
+        explanation
+    optional: (optional) string
+        A string specifying optional variable
+        e.g. strings
+
+    Returns
+    -------
+    output: tuple
+    """
+    pass
+# think: how to test
+
+def read_genx(path_to_file):
+    """
+    This definition reads the .genx output file from SolarSoft Ware (SSW). It will extract the instrument data and save them to an data file to be used for this program.
+
+     Parameters
+    ----------
+    :param path_to_file: string, the path location that leads to ssw_aia_response_genx file documents. The output file
+                    used for obtaining all of the information is:   aia_V6_all_fullinst
+    :param outfile_name: string, the name of the outfile that will store the dataframe
+                                    - could make this a dictioary called self.instr_properties in class format
+
+    Returns
+    -------
+    output: a comma separated dataframe file with arrays for each property per channel
+
+    """
+
+    # access outer dictionary from .genx file
+    ssw_dict = readsav(path_to_file)
```

Comment:
@dpshelio As of now, this function only reads IDL save files (that is, `.genx` files read into IDL via `restgen.pro` and then saved as IDL save files). I'm guessing you're trying to read in a normal `.genx` file. It seems that `scipy.io.readsav` cannot read these. This is of course not an ideal or permanent solution.